x/arkeo/types/keeper.go

```diff
@@ -32,60 +32,99 @@ func NewContract(provider common.PubKey, chain common.Chain, client common.PubKe
 	}
 }
 
-func (c Contract) Key() string {
-	return strconv.FormatUint(c.Id, 10)
+func (contract Contract) Key() string {
+	return strconv.FormatUint(contract.Id, 10)
 }
 
-func (c Contract) GetSpender() common.PubKey {
-	if !c.Delegate.IsEmpty() {
-		return c.Delegate
+func (contract Contract) GetSpender() common.PubKey {
```

Comment:
usually people like to use a very short variable , because no one likes to type the full name :) 

but I am cool you to change it

Reply:
auto-complete and readable variable names are a great combo :)